website/source/api/auth/alicloud/index.html.md

```diff
@@ -0,0 +1,225 @@
+
+This documentation assumes the AliCloud method is mounted at the `/auth/alicloud`
+path in Vault. Since it is possible to enable auth methods at any location,
+please update your API calls accordingly.
+
+## Create Role
+
+Registers a role. Only entities using the role registered using this endpoint 
+will be able to perform the login operation.
+
+| Method   | Path                             | Produces               |
+| :------- | :------------------------------- | :--------------------- |
+| `POST`   | `/auth/alicloud/role/:role`      | `204 (empty body)`     |
+
+### Parameters
+
+- `role` `(string: <required>)` - Name of the role. Must correspond with the name of the role reflected in the arn.
+- `arn` `(string: <required>)` - The role's arn.
+- `policies` `(array: [])` - Policies to be set on tokens issued using this
+  role.
+- `ttl` `(string: "")` - The TTL period of tokens issued using this role,
+  provided as "1h", where hour is the largest suffix.
+- `max_ttl` `(string: "")` - The maximum allowed lifetime of tokens issued using
+  this role.
+- `period` `(string: "")` - If set, indicates that the token generated using
+  this role should never expire. The token should be renewed within the duration
+  specified by this value. At each renewal, the token's TTL will be set to the
+  value of this parameter.
+- `bound_cidrs` `(string: "", or list: [])` – If set, restricts usage of the
+  roles to client IPs falling within the range of the specified CIDR(s).
+
+### Sample Payload
+
+```json
+{
+  "arn": "acs:ram::5138828231865461:role/elk",
+  "policies": [
+    "dev",
+    "prod"
+  ]
+}
+```
+
+### Sample Request
+
+```
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    --data @payload.json \
+    http://127.0.0.1:8200/v1/auth/alicloud/role/elk
+```
+
+## Read Role
+
+Returns the previously registered role configuration.
+
+| Method   | Path                         | Produces               |
+| :------- | :--------------------------- | :--------------------- |
+| `GET`    | `/auth/alicloud/role/:role`  | `200 application/json` |
+
+### Parameters
+
+- `role` `(string: <required>)` - Name of the role.
+
+### Sample Request
+
+```
+$ curl \
+    --header "X-Vault-Token: ..." \
+    http://127.0.0.1:8200/v1/auth/alicloud/role/elk
+```
+
+### Sample Response
+
+```json
+{
+  "data": {
+    "arn": "acs:ram::5138828231865461:role/elk",
+    "policies": [
+      "default",
+      "dev",
+      "prod"
+    ],
+    "ttl": 1800000,
+    "max_ttl": 1800000,
+    "period": 0
+  }
+}
+```
+
+## List Roles
+
+Lists all the roles that are registered with the method.
+
+| Method   | Path                         | Produces               |
+| :------- | :--------------------------- | :--------------------- |
+| `LIST`   | `/auth/alicloud/roles`       | `200 application/json` |
+
+### Sample Request
+
+```
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request LIST \
+    http://127.0.0.1:8200/v1/auth/alicloud/roles
```

Comment:
I will add `?list=true` to the end of this URL.

Reply:
We've been explicitly excluding these from the docs.  the `--request LIST` will take care of this.